### ISSUE TITLE
[Bug]: Apple M1 / .NET 5 playwright command fails to start

### ISSUE THREAD

**Issue by dhsl** [microsoft/playwright-dotnet#1836]
### Playwright version

1.17.0

### Operating system

MacOS

### Other information

```
❯ dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.402
 Commit:    e9d3381880

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  11.0
 OS Platform: Darwin
 RID:         osx-x64
 Base Path:   /usr/local/share/dotnet/sdk/5.0.402/

Host (useful for support):
  Version: 5.0.11
  Commit:  f431858f8b

.NET SDKs installed:
  5.0.402 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
```

```
❯ file /usr/local/share/dotnet/host/fxr/5.0.11/libhostfxr.dylib
/usr/local/share/dotnet/host/fxr/5.0.11/libhostfxr.dylib: Mach-O 64-bit dynamically linked shared library x86_64
❯ file ~/.dotnet/tools/playwright
/Users/dhe/.dotnet/tools/playwright: Mach-O 64-bit executable x86_64
```

### What happened? / Describe the bug

After cleanly installing .net 5 and Playwright.CLI 1.17.0 globally on my M1 Apple machine, the playwright command fails to start complaining about missing **libhostfxr.dylib**.

The file seems to exist in **/usr/local/share/dotnet/host/fxr/5.0.11/libhostfxr.dylib** and is accessible by the user.

The playwright executable and the libhostfxr.dylib both are compiled for x86_64.

Not sure if it has something to do with /etc/dotnet/install_location points to /usr/local/share/dotnet/x64 which does not exist.

### Relevant log output

```shell
❯ playwright
A fatal error occurred. The required library libhostfxr.dylib could not be found.
If this is a self-contained application, that library should exist in [/Users/dhe/.dotnet/tools/.store/microsoft.playwright.cli/1.2.0/microsoft.playwright.cli/1.2.0/tools/net5.0/any/].
If this is a framework-dependent application, install the runtime in the global location [/usr/local/share/dotnet/x64] or use the DOTNET_ROOT environment variable to specify the runtime location or register the runtime location in [/etc/dotnet/install_location].

The .NET runtime can be found at:
  - https://aka.ms/dotnet-core-applaunch?missing_runtime=true&arch=x64&rid=osx.11.1-x64&apphost_version=5.0.11
```

**Comment by dhsl** [comment]
Found a workaround (at least for CLI for now) by setting DOTNET_ROOT to /usr/local/share/dotnet when running playwright.

**Comment by programatt** [comment]
@dhsl for me setting DOTNET_ROOT does nothing to change the error.

```
% DOTNET_ROOT=/usr/local/share/dotnet playwright install
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '5.0.0' (arm64) was not found.
  - The following frameworks were found:
      6.0.0 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=5.0.0&arch=arm64&rid=osx.11.1-arm64
```

**Comment by dhsl** [comment]
> @dhsl for me setting DOTNET_ROOT does nothing to change the error.

@programatt Do you have .NET 5 installed? The error message suggests you only have .NET 6 installed.

**Comment by programatt** [comment]
@dhsl I do, and i've done the `/etc/dotnet/install_location` edit as well.

**Comment by dhsl** [comment]
@programatt, I know. I'm on a M1 machine running .NET 5 which does not have a port for arm64, so it runs the x64.
In this constellation I get the above error when trying to run playwright. It works when I do **DOTNET_ROOT=/usr/local/share/dotnet playwright** instead.

I know that .NET 6 has a native version but I'm not running that since the project is bound to .NET 5 and not .NET 6.

Your error is different, since you only have .NET 6 installed and not .NET 5 ;)

**Comment by programatt** [comment]
@dhsl. I actually do have .net 5 installed. I'm not sure why your output is different than mine. I am aware there is no .NET 5 arm64 build of the sdk. But I did download .net 5 x64 and .net 6 arm64 and installed both of them. I also tried setting up my machine as specified https://github.com/dotnet/sdk/issues/20584 but I still get the above error even setting DOTNET_ROOT. I feel kinda lost with the *magic* to get it to work.

**Comment by dhsl** [comment]
@programatt try to uninstall .NET 6. There are some side effects when you have both of them installed. Looks like it chooses .NET 6 (which is arm64) and doesn't show .NET 5 there.

I use this script (found it somewhere inside the microsoft universe) to remove all .NET components before installing the SDK I want to use, which in this case is only .NET 5.

```
#!/usr/bin/env bash
#
# Copyright (c) .NET Foundation and contributors. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project root for full license information.
#

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"

current_userid=$(id -u)
if [ $current_userid -ne 0 ]; then
    echo "$(basename "$0") uninstallation script requires superuser privileges to run" >&2
    exit 1
fi

# this is the common suffix for all the dotnet pkgs
dotnet_pkg_name_suffix="com.microsoft.dotnet"
dotnet_install_root="/usr/local/share/dotnet"
dotnet_path_file="/etc/paths.d/dotnet"
dotnet_tool_path_file="/etc/paths.d/dotnet-cli-tools"

remove_dotnet_pkgs(){
    installed_pkgs=($(pkgutil --pkgs | grep $dotnet_pkg_name_suffix))

    for i in "${installed_pkgs[@]}"
    do
        echo "Removing dotnet component - \"$i\"" >&2
        pkgutil --force --forget "$i"
    done
}

remove_dotnet_pkgs
[ "$?" -ne 0 ] && echo "Failed to remove dotnet packages." >&2 && exit 1

echo "Deleting install root - $dotnet_install_root" >&2
rm -rf "$dotnet_install_root"
rm -f "$dotnet_path_file"
rm -f "$dotnet_tool_path_file"

echo "dotnet packages removal succeeded." >&2
exit 0
```

**Comment by programatt** [comment]
@dhsl Thanks but I don't really want to remove .net 6.0 since i'm actively using it for other things including a PR here for upgrading it to .net 6. I did finally find that setting `DOTNET_ROLL_FORWARD=Major` in my shell allowed it to work. But that doesn't seem like the correct long term solution.

**Comment by mxschmitt** [comment]
This should be fixed in version 1.2.1 of Microsoft.Playwright.CLI.
